assembly/src/assembler/instruction/mod.rs

```diff
@@ -306,6 +306,15 @@ impl Assembler {
             Instruction::CallLocal(idx) => self.call_local(*idx, ctx),
             Instruction::CallImported(id) => self.call_imported(id, ctx),
             Instruction::SysCall(id) => self.syscall(id, ctx),
+
+            // ----- debug decorators -------------------------------------------------------------
+            Instruction::Breakpoint => {
+                if self.in_debug_mode() {
+                    span.add_op(Noop)?;
+                    span.track_instruction(instruction, ctx);
+                }
+                Ok(None)
```

Comment:
If we fail here, then we will fail whenever `breakpoint` exists as MASM instruction. Here we check `in_debug_mode`, and add the `Noop` only if in debug mode.

Can you elaborate?

Reply:
Would that be convenient tho? We would force the user to add and remove these instructions, constantly.

To be fair, this is not really optimal as ideally we should have a source map and the ability to insert breakpoints with the debugger.

Reply:
wdyt @bobbinth @grjte ?

Reply:
The main challenge is how the vm state iterator works. It will iterate ticks of the clock (hence, ops).

The way it is done now is to avoid a bigger impact, that would be to have the vm state iterator iterate through decorators.

With the current implementation, we break when we have a `noop` (or any other opcode) decorated with the boolean flag `should_break`